src/common/arrow-ffi/src/lib.rs

```diff
@@ -37,21 +40,25 @@ pub fn to_py_array<'py>(
     array: ArrayRef,
     pyarrow: &Bound<'py, PyModule>,
 ) -> PyResult<Bound<'py, PyAny>> {
-    let schema = Box::new(ffi::export_field_to_c(&Field::new(
+    let field = arrow::ffi::FFI_ArrowSchema::try_from(&arrow_schema::Field::new(
         "",
-        array.data_type().clone(),
+        array.data_type().clone().into(),
```

Comment:
Is this into fallible? How does extension types convert correctly, since they are DataTypes in arrow2 but fields in arrow-rs

Reply:
We use arrow extension types to ensure that our custom Daft types are round-trippable to arrow. We just need to make sure that this is still the case here. With arrow-rs, as Srinu said, extension datatype metadata is lost, so we'll need to find a way to preserve it.

Reply:
@srilman @kevinzwang I updated this PR so that extensions are preserved during round trips. the previously failing round trip tests are now passing.